bamwallet/HD/WalletService.cs

```
@@ -964,21 +965,28 @@ public async Task<string[]> CreateSeed(Language language, WordCount wordCount)
         /// <returns></returns>
         public TaskResult<IEnumerable<string>> WalletList()
         {
-            var wallets = Path.Combine(Path.GetDirectoryName(AppDomain.CurrentDomain.BaseDirectory) ?? string.Empty,
-                "wallets");
-
-            string[] files;
+            var baseDir = Path.GetDirectoryName(AppDomain.CurrentDomain.BaseDirectory);
```

Instead of returning an empty string, this function seems to throw an exception.

---

Before returning an empty string + "wallets" would guaranteed throw an exception. The only way that I know of that this will throw is to unload the domain, or not have a loaded domain. If you can reproduce this somehow let me know :)